test/integration/activate_int_test.go

```diff
@@ -217,14 +218,36 @@ func (suite *ActivateIntegrationTestSuite) activatePython(version string, extraE
 	cp.SendLine("exit")
 	cp.ExpectExitCode(0)
 
+	executor := filepath.Join(ts.Dirs.DefaultBin, pythonShim)
 	// check that default activation works
 	cp = ts.SpawnCmdWithOpts(
-		filepath.Join(ts.Dirs.DefaultBin, pythonShim),
+		executor,
 		e2e.WithArgs("-c", "import sys; print(sys.copyright);"),
 		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
 	)
 	cp.Expect("ActiveState Software Inc.")
 	cp.ExpectExitCode(0)
+
+	// check that default activation does not recurse
+	cp = ts.SpawnCmdWithOpts(
+		executor,
+		e2e.WithArgs("-c", fmt.Sprintf(`import subprocess; subprocess.run(["%s", "-c", "print('hello')"])`, executor)),
+		e2e.AppendEnv("ACTIVESTATE_CLI_DISABLE_RUNTIME=false"),
+	)
```

Comment:
I surely may be testing incorrectly; I don't think this represents the recursion issue when VSCode is involved. I've tested this locally without success.

Reply:
Scratch this. Somehow I was on the wrong commit. It does seem to work.